src/lib/Libifl/pbsD_msgjob.c

```diff
@@ -202,6 +202,50 @@ char *extend;
 	if (pbs_client_thread_init_thread_context() != 0)
 		return pbs_errno;
 
+	/* first verify the resource list in keep_select option */
+	if (extend) {
+		struct attrl *attrib = NULL;
+		char ebuff[200], *erp, *emsg = NULL;
+		int i;
+		struct pbs_client_thread_connect_context *con;
+		char nd_ct_selstr[20];
+		char *endptr = NULL;
+
+		strtol(extend, &endptr, 10);
+
+		if (!*endptr) {
+			snprintf(nd_ct_selstr, sizeof(nd_ct_selstr), "select=%s", extend);
+			extend = nd_ct_selstr;
+		} else if ((i = set_resources(&attrib, extend, 1, &erp))) {
+			if (i > 1) {
+				snprintf(ebuff, sizeof(ebuff), "illegal -k value: %s\n", pbs_parse_err_msg(i));
+				emsg = strdup(ebuff);
+			} else
+				emsg = strdup("illegal -k value\n");
+			pbs_errno = PBSE_INVALSELECTRESC;
+		} else {
+			if (!attrib || strcmp(attrib->resource, "select")) {
+				emsg = strdup("only a \"select=\" string is valid in -k option\n");
+				pbs_errno = PBSE_IVALREQ;
+			} else
+				pbs_errno = PBSE_NONE;
+		}
+		if (pbs_errno) {
+			if ((con = pbs_client_thread_find_connect_context(c))) {
+				if (con->th_ch_errtxt)
+					free(con->th_ch_errtxt);
+				con->th_ch_errtxt = emsg;
+				con->th_ch_errno = pbs_errno;
+			} else
+				connection[c].ch_errtxt = emsg;
```

Comment:
If c is the connection index passed in, why did we call pbs_client_thread_find_connect_context() to find it?
I assume the storage for emsg is free()'d when the connection is cleaned up?
Why don't we set connection[c].ch_errno here?

Reply:
the call to `pbs_client_thread_find_connect_context()` is to find `con` which is of type `struct pbs_client_thread_connect_context *`.
Yes, emsg is free()'d when the connection is cleaned up.
Frankly after erroring out from here, there's nobody reading `connection[c].ch_errno`. But anyways I added it in the commit https://github.com/PBSPro/pbspro/pull/1452/commits/8f309272ad00f7587c830cad07238cb40598e24a for completeness.